src/main/java/com/endava/internship/collections/StudentList.java

```diff
@@ -1,140 +1,399 @@
 package com.endava.internship.collections;
 
+
 import java.util.Collection;
 import java.util.Iterator;
 import java.util.List;
+import java.util.Objects;
+import java.util.Arrays;
 import java.util.ListIterator;
 
+
 public class StudentList implements List<Student> {
 
+    private int capacity;
+
+    private int size = 0;
+
+    private Student[] array;
+
+
+    public StudentList() {
+        this.capacity = 10;
+        this.array = new Student[this.capacity];
+    }
+
+    public StudentList(int initialCapacity) {
+        if (initialCapacity >= 0) {
+            this.capacity = initialCapacity;
+            this.array = new Student[this.capacity];
+        } else {
+            if (initialCapacity < 0) {
+                throw new IllegalArgumentException("Initial capacity must be greater than 0 : initialCapacity = " + initialCapacity);
+            }
+
+            this.capacity = 10;
```

Comment:
This code never will be invoked

Reply:
Fixed

Reply:
I suggest you to extract some cases
`
if(initialCapacity >=0) {
this.capacity = initialCapacity;
this.array = new Student[initialCapacity];
}else{
//... throw exception
}
}